### ISSUE TITLE
policy required: Xcode15 requires Privacy Manifest

### ISSUE THREAD

**Issue by yaofeijian** [braintree/braintree_ios#1135]
### Is your feature request related to a problem? Please describe.

Apple review requires applecations to include a privacy manifest files, all submissions that don't support privacy will be rejected, ref link:https://developer.apple.com/documentation/bundleresources/privacy_manifest_files
when will Braintree update the SDK

### Describe the solution you'd like.

please check the link.
https://developer.apple.com/documentation/bundleresources/privacy_manifest_files/describing_data_use_in_privacy_manifests

**Comment by jaxdesmarais** [comment]
Hello @yaofeijian -

Thanks for reaching out and using the SDK! We are aware of Apple's new privacy manifest requirements and plan to make the necessary changes early next year ahead of the April 2024 deadline. Additionally we are working with our 3rd party SDKs to ensure that they have meet the requirements as well. We will keep this issue open with the backlog tag until this work has been completed so other folks in the community can stay updated as well.

**Comment by scannillo** [comment]
👋 Hi folks - wanted to update you all that we are continuing to work with our SDK partners to deliver us their privacy manifest files in time ahead of the April 2024 deadline.

**Comment by DanielReyesDev** [comment]
Hi @jaxdesmarais do we know the timelines about adding the Privacy Manifest and Code Signature to the framework? Thanks in advance!

**Comment by NiallSmyth** [comment]
We have received notice from PayPal:

> It is necessary for all third-party apps to integrate the latest Magnes 5.5.0 version by March 19 2024 in order to successfully push updates to the Apple app store.
App deployments with an older version of Magnes SDK (5.4.1 or lesser) will not be accepted after this date.
Please contact the Merchant to ensure they have upgraded their Magnes integration already. In case of any technical difficulties please reach out to the integration team for support.

This date is only 2 weeks away, but Braintree iOS is still using Magnes SDK 5.4.1. @scannillo Can you put a date on when this SDK will be updated, or do we need to look at moving away from Braintree iOS and back to including the Magnes framework directly in our app?

**Comment by scannillo** [comment]
Hi @NiallSmyth - we plan to update the Braintree SDK to use the v5.5.0 of Magnes by end of this week, early next!

Also, where did you receive that notification from Magnes?

**Comment by NiallSmyth** [comment]
Thanks for the update @scannillo. We were emailed directly by our PayPal account managers about the required update to Magnes.

**Comment by RGG-jayoung** [comment]
We are also blocked by this issue, will version 5.x and 6.x be updated with the privacy manifest and Magnes changes? Any update on timing would also be appreciated so we can coordinate QA internally.

<img width="1159" alt="image" src="https://github.com/braintree/braintree_ios/assets/53495188/452ecf4f-72d5-49a2-88d6-b201fc147a19">

**Comment by NiallSmyth** [comment]
@RGG-jayoung that looks like an unrelated issue which is actually caused by building with Xcode 15.3. The Firebase SDK has [a workaround](https://github.com/firebase/firebase-ios-sdk/commit/4065853eef0c5c5c09f59107aec14e0affeb8659) for the same issue. If you build with Xcode 15.2 it should work.

**Comment by RGG-jayoung** [comment]
> @RGG-jayoung that looks like an unrelated issue which is actually caused by building with Xcode 15.3. The Firebase SDK has [a workaround](https://github.com/firebase/firebase-ios-sdk/commit/4065853eef0c5c5c09f59107aec14e0affeb8659) for the same issue. If you build with Xcode 15.2 it should work.

Thank you @NiallSmyth for the quick follow up. We'll try submissions with 15.2 for the time being. 

If it helps and is specific to Braintree, Firebase seems to have solved for this in their [latest release](https://github.com/firebase/firebase-ios-sdk/releases/tag/10.22.1). The only hold out is the error pictured above, flagging flag `PPRiskMagnes.framework`. Searching on line kept landing us here.

Thank you again for your help.

**Comment by scannillo** [comment]
Hi @RGG-jayoung - we have a PR up for moving to Magnes v5.5.0 for Braintree iOS V6 (https://github.com/braintree/braintree_ios/pull/1217)

**Comment by RGG-jayoung** [comment]
> Hi @RGG-jayoung - we have a PR up for moving to Magnes v5.5.0 for Braintree iOS V6 (#1217)

@scannillo thank you for this update! Is this also possible within the v5 releases?

**Comment by scannillo** [comment]
We just release [v6.14.0](https://github.com/braintree/braintree_ios/releases/tag/6.14.0) with Magnes 5.5.0. We are working on getting a version of Magnes compatible with Braintree iOS V5.

**Comment by mammaiap** [comment]
when will be the new version going to be released with "Privacy Manifest" feature?.

**Comment by KunJeongPark** [comment]
Hello, we are doing a release with the "Privacy Manifest" feature today.

**Comment by KunJeongPark** [comment]
We are working on release with the Privacy Manifest files today. Thank you.

**Comment by dmazzoni** [comment]
Are you also planning an update to v5 with the privacy manifest? We are using the Drop-In SDK and therefore cannot update to v6.

**Comment by ken0nek** [comment]
@dmazzoni I think the Braintree team kindly supports v5 too!
https://github.com/braintree/braintree_ios/releases/tag/5.25.0

**Comment by jaxdesmarais** [comment]
Hello - 

Yes we have updated v5 and will be updating the Drop-in with the latest version of v5 by the end of this week. Thanks again for your patience as we roll out the privacy manifest updates!

**Comment by mehal** [comment]
So is v6.17 all set for Xcode?  We're still getting a Test Flight validation error.

**Comment by jaxdesmarais** [comment]
Hey @mehal -

Version 6.17.0 should include all privacy manifests except for our dependency in `PayPalNativeCheckout` as we have not yet received an updated version from that team. Since that module receives a small amount of traffic we did not want to block merchants using other modules since that dependency has not yet provided a timeline on this change. 

Would you mind sharing the validation error you are seeing and any other details to help us troubleshoot? If the error is unrelated to privacy manifests, please open a new issue with those details.

**Comment by mehal** [comment]
OK, I'll get the dev working on this to give me details.

**Comment by mehal** [comment]
Hi @jaxdesmarais

Hi Jax, looks like I was wrong as we were able to get past our error.  Miscommunication on our end. 

Thanks for the help - Mike

**Comment by scannillo** [comment]
Please upgrade to the latest version of our SDK, which satisfies Apple's privacy manifest requirements for SDKs (added in [v6.15.0](https://github.com/braintree/braintree_ios/blob/main/CHANGELOG.md#6150-2024-03-18))
